hierarchy.el

```diff
@@ -534,11 +563,20 @@ value (a number) as parameter and inserting a string to be displayed as a
 node label."
   (require 'wid-edit)
   (require 'tree-widget)
-  (hierarchy-map-tree (lambda (item indent children)
-                        (widget-convert
-                         'tree-widget
-                         :tag (hierarchy-labelfn-to-string labelfn item indent)
-                         :args children))
+  (hierarchy-map-tree (lambda (item indent childrenfn children)
+                        (if childrenfn
+                            (widget-convert
+                             'tree-widget
+                             :tag (hierarchy-labelfn-to-string labelfn item indent)
+                             :expander (hierarchy--create-delayed-tree-widget
```

Comment:
I couldn't find any documentation for `:expander`. Can you tell me where to read about it?

Reply:
Yeah; their documentation is, really, not the best.

I found it as a comment in their source code: https://github.com/emacs-mirror/emacs/blob/master/lisp/tree-widget.el. Just do a search for `:expander`.

Basically, you can provide a function to get dynamic results when the widget is expanded (when the children of the widget are asked for), so long as the function provides a list of what should serve as the children.